packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol

```diff
@@ -363,7 +366,13 @@ contract BridgeFacet is BaseConnextFacet {
     {
       // Check that the asset is supported -- can be either adopted or local.
       TokenId memory canonical = s.adoptedToCanonical[_args.transactingAsset];
-      if (canonical.id == bytes32(0)) {
+
+      // NOTE: above we check that you can only have `address(0)` as a transacting asset when
+      // you are sending 0-amounts. Because 0-amount transfers shortcircuit all checks on
+      // mappings keyed on hash(canonicalId, canonicalDomain), this is safe even when the
+      // address(0) asset is not whitelisted. These values are only used for the `transactionId`
+      // generation
+      if (canonical.id == bytes32(0) && _args.transactingAsset != address(0)) {
```

Comment:
what happens if the _args.transactingAsset == address(0) ?? should we still be calling `transferAssetToContract` and `swapToLocalAssetIfNeeded` below?

Reply:
oh, I see, we do nothing in that method if the amount is 0... but still, we should just include the whole "transfer asset to contract and do asset stuff" step within a larger conditional block that checks to make sure this isn't a transfer with `_args.transactingAsset == address(0)`... at least, that seems like the more readable and efficient way to do it imo

Reply:
reasonable